homeassistant/components/overkiz/cover.py

```diff
@@ -1,40 +1,413 @@
+                OverkizCover(
+                    device.device_url,
+                    data.coordinator,
+                    description,
+                )
+            )
 
-    entities += [
-        VerticalCover(device.device_url, data.coordinator)
-        for device in data.platforms[Platform.COVER]
-        if device.ui_class != UIClass.AWNING
-    ]
+    async_add_entities(entities)
 
-    entities += [
-        LowSpeedCover(device.device_url, data.coordinator)
-        for device in data.platforms[Platform.COVER]
-        if OverkizCommand.SET_CLOSURE_AND_LINEAR_SPEED in device.definition.commands
-    ]
 
-    async_add_entities(entities)
+class OverkizCover(OverkizDescriptiveEntity, CoverEntity):
+    """Representation of an Overkiz Cover."""
+
+    entity_description: OverkizCoverDescription
+
+    def __init__(
+        self,
+        device_url: str,
+        coordinator: OverkizDataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize the device."""
+        super().__init__(device_url, coordinator, description)
+
+        # Use device url as unique ID for backwards compatibility
+        self._attr_unique_id = self.device.device_url
+
+        # Overkiz does support covers where only tilt commands are supported
+        # and HA sets by default open/close as supported feature which conflicts
+        supported_features = CoverEntityFeature(0)
+
+        if self.entity_description.open_command:
+            supported_features |= CoverEntityFeature.OPEN
+
+            if self.entity_description.stop_command:
+                supported_features |= CoverEntityFeature.STOP
+
+        if self.entity_description.close_command:
+            supported_features |= CoverEntityFeature.CLOSE
+
+        if self.entity_description.open_tilt_command:
+            supported_features |= CoverEntityFeature.OPEN_TILT
+
+            if self.entity_description.stop_tilt_command:
+                supported_features |= CoverEntityFeature.STOP_TILT
+
+        if self.entity_description.close_tilt_command:
+            supported_features |= CoverEntityFeature.CLOSE_TILT
+
+        if self.entity_description.set_tilt_position_command:
+            supported_features |= CoverEntityFeature.SET_TILT_POSITION
+
+        if self.entity_description.set_position_command:
+            supported_features |= CoverEntityFeature.SET_POSITION
+
+        self._attr_supported_features = supported_features
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return if the cover is closed."""
+        if is_closed_fn := self.entity_description.is_closed_fn:
+            return is_closed_fn(self.device)
+
+        return None
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        state_name = self.entity_description.current_position_state
```

Comment:
The vertical_cover was also using OverkizState.CORE_CLOSURE_OR_ROCKER_POSITION, OverkizState.CORE_PEDESTRIAN_POSITION.  
These 2 states have been completly removed from the current code.
Is it intended and known that's it's not useful anymore?